### PR TITLE
Method index

### PR DESCRIPTION
**Pull request by LuisARodr**
This adds the `index(spec)` method functionality to the framework.
Also includes tests for the method.

Description of the method:
>Will return the zero based position in the array of the element that satisfies the specification. The specification can be either an object value or a function. If no element  match the specification then a -1 value will be returned.

### CODE REVIEW

File: Methods/ArrayExtension.js [hunk 1]
```diff
@@ -224,6 +224,35 @@ Array.prototype.count = function(spec) {
     return count;
 }
 
+/**
+ * Return the zero based position in the array of the element that satisfies
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js]: Returns*

File: Methods/ArrayExtension.js [hunk 2]
```diff
@@ -224,6 +224,35 @@ Array.prototype.count = function(spec) {
...
+ * spec.
+ * @param {spec|*} spec - The value to be matched on every value of the array 
+ * on search of it, if spec is a fuction then a value is found when the function
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js]: function*

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    it('retunrs 3 when the 4th value is a match with a function spec', function() {
+      const array = [1, 2, 3, 4];
+
+      const valueIndex = array.index(value => value > 3);
+
+      expect(valueIndex).to.be.a('number').that.equal(3);
+    });
+
+    it("returns -1 when there's no match for the with an obj spec", function() {
```
Comment (gustavostuff) [inline test/index.js]: phrasing

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      const array = [1, 2, 3, 4];
+
+      const valueIndex = array.index(5);
+
+      expect(valueIndex).to.be.a('number').that.equal(-1);
+    });
+
+    it("returns -1 when there's no match for the with a function spec", function() {
```
Comment (gustavostuff) [inline test/index.js]: same as above

**Review by gustavostuff — CHANGES_REQUESTED** [review]

**Review by gustavostuff — APPROVED** [review]

**Comment by LuisARodr** [comment]
Closing to merge into [dev](https://github.com/LuisARodr/array-extension/tree/dev)

**Pull request closed without merging**